usvm-jvm/src/main/kotlin/org/usvm/machine/JcApproximations.kt

```diff
@@ -110,66 +85,84 @@ class JcMethodApproximationResolver(
     }
 
     private fun approximateRegularMethod(methodCall: JcMethodCall): Boolean = with(methodCall) {
-        if (method.enclosingClass == usvmApiSymbolicList) {
+        val enclosingClass = method.enclosingClass
+        val className = enclosingClass.name
+
+        if (enclosingClass == usvmApiSymbolicList) {
             approximateUsvmSymbolicListMethod(methodCall)
             return true
         }
 
-        if (method.enclosingClass == usvmApiSymbolicMap) {
+        if (enclosingClass == usvmApiSymbolicMap) {
             approximateUsvmSymbolicMapMethod(methodCall)
             return true
         }
 
-        if (method.enclosingClass == usvmApiSymbolicIdentityMap) {
+        if (enclosingClass == usvmApiSymbolicIdentityMap) {
             approximateUsvmSymbolicIdMapMethod(methodCall)
             return true
         }
 
-        if (method.enclosingClass == ctx.cp.objectClass) {
+        if (enclosingClass == ctx.cp.objectClass) {
             if (approximateObjectVirtualMethod(methodCall)) return true
         }
 
-        if (method.enclosingClass == ctx.classType.jcClass) {
+        if (enclosingClass == ctx.classType.jcClass) {
             if (approximateClassVirtualMethod(methodCall)) return true
         }
 
-        if (method.enclosingClass.name == "jdk.internal.misc.Unsafe") {
+        if (className == "jdk.internal.misc.Unsafe") {
             if (approximateUnsafeVirtualMethod(methodCall)) return true
         }
 
-        if (method.name == "clone" && method.enclosingClass == ctx.cp.objectClass) {
+        if (method.name == "clone" && enclosingClass == ctx.cp.objectClass) {
             if (approximateArrayClone(methodCall)) return true
         }
 
+        if (className == "org.apache.commons.logging.Log") {
+            if (approximateLoggerMethod(methodCall)) return true
+        }
+
         return approximateEmptyNativeMethod(methodCall)
     }
 
     private fun approximateStaticMethod(methodCall: JcMethodCall): Boolean = with(methodCall) {
-        if (method.enclosingClass == usvmApiEngine) {
+        val enclosingClass = method.enclosingClass
+        val className = enclosingClass.name
+        if (enclosingClass == usvmApiEngine) {
             approximateUsvmApiEngineStaticMethod(methodCall)
             return true
         }
 
-        if (method.enclosingClass == ctx.classType.jcClass) {
+        if (enclosingClass == ctx.classType.jcClass) {
             if (approximateClassStaticMethod(methodCall)) return true
         }
 
-        if (method.enclosingClass.name == "java.lang.System") {
+        if (className == "generated.java.lang.PrimitiveTypeUtils") {
```

Comment:
Approximation for the approximation?)

Reply:
Да, Class.getPrimitiveClass является приватным методом, поэтому в аппроксимациях примитивных классов используется PrimitiveTypeUtils

Reply:
Maybe we can write an approximation for `Class.getPrimitiveClass`?  I don't like that we have dependency on approximations implementation in USVM. So, i want to avoid `generated.java.lang.PrimitiveTypeUtils` here.

Reply:
We already have approximation for Class.getPrimitiveClass